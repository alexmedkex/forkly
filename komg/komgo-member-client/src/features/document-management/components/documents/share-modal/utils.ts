import { ICompanyTableItemWithSharedDocs, ICounterpartyWithDocuments } from './SelectedCounterparties'
import { FlagNameValues } from 'semantic-ui-react'
import * as i18nIsoCountries from 'i18n-iso-countries'

export function pluralize(word: string, count: number, singleSuffix: string, pluralSuffix: string) {
  if (count > 1) {
    return `${word}${pluralSuffix}`
  }

  return `${word}${singleSuffix}`
}

export const buildCounterpartyWithDocumentsPickerItems = (
  counterpartiesWithDocs: ICounterpartyWithDocuments[]
): ICompanyTableItemWithSharedDocs[] => {
  i18nIsoCountries.registerLocale(require('i18n-iso-countries/langs/en.json'))
  return counterpartiesWithDocs.map(item => ({
    name: item.counterparty.x500Name.CN,
    countryName: i18nIsoCountries.getName(item.counterparty.x500Name.C.trim(), 'en'),
    country: item.counterparty.x500Name.C.toLocaleLowerCase().trim() as FlagNameValues,
    location: item.counterparty.x500Name.L,
    id: item.counterparty.staticId,
    sharedDocuments: item.documents
  }))
}
