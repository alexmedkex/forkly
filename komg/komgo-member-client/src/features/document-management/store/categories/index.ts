import reducer from './reducer'

import * as CategoryActions from './actions'

export { reducer as CategoriesReducer, CategoryActions }
