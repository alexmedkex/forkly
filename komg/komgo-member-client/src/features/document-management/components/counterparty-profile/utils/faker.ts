import { CounterpartyProfile, RiskLevel } from '../../../../counterparties/store/types'

const defaultProfile: CounterpartyProfile = {
  id: 'anon-profile-id',
  counterpartyId: 'anon-counterparty-id',
  renewalDate: new Date('2019-07-16').toISOString(),
  riskLevel: RiskLevel.unspecified,
  managedById: 'anon-user-id'
}

export const fakeProfile = (partial?: Partial<CounterpartyProfile>) => {
  return partial ? { ...defaultProfile, ...partial } : defaultProfile
}

export const mockDate = () => {
  const RealDate = Date
  const _GLOBAL: any = global // we love typescript!

  return {
    freeze: (isoDate: string | number | Date) => {
      _GLOBAL.Date = class extends RealDate {
        constructor(...args: any) {
          super()
          return new RealDate(isoDate)
        }
      }
    },
    restore: () => {
      global.Date = RealDate
    }
  }
}
