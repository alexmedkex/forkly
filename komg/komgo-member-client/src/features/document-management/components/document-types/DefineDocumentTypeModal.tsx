import * as React from 'react'
import { Form, Modal, Button, Input, Dropdown } from 'semantic-ui-react'
import { Category } from '../../store/types'

export interface Props {
  visible: boolean
  title: string
  categories: Category[]
  predefinedData: { category: string; name: string; id?: string }
  toggleVisible(): void
  onCreateSuccess(cat: string, name: string): void
  onEditSuccess(cat: string, name: string, id: string): void
}

interface State {
  category: string
  name: string
}

interface CategoryRepresentation {
  key: string
}

const formatCategoriesToSelect = (categories: Category[]): CategoryRepresentation[] => {
  return categories.map((cat: Category) => {
    return { key: cat.id, text: cat.name, value: cat.id }
  })
}

class DefineDocumentTypeModal extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props)
  }

  handleSubmit = () => {
    if (this.props.predefinedData.id) {
      this.props.onEditSuccess(
        this.state.category ? this.state.category : this.props.predefinedData.category,
        this.state.name,
        this.props.predefinedData.id
      )
    } else {
      this.props.onCreateSuccess(this.state.category, this.state.name)
    }
  }

  render() {
    return (
      <Modal open={this.props.visible} size="mini">
        <Modal.Header>{this.props.title}</Modal.Header>
        <Modal.Content>
          <Modal.Description>
            <Form onSubmit={this.handleSubmit} id="submit-form">
              <Form.Field>
                <label>Category</label>
                <Dropdown
                  placeholder="Select category"
                  defaultValue={
                    this.props.predefinedData.category !== '' ? this.props.predefinedData.category : undefined
                  }
                  options={formatCategoriesToSelect(this.props.categories)}
                  onChange={this.setSelectedCategoryId}
                />
              </Form.Field>
              <Form.Field>
                <label>Name</label>
                <Input
                  placeholder="My new doctype"
                  defaultValue={this.props.predefinedData.name !== '' ? this.props.predefinedData.name : undefined}
                  onChange={(e: any) => {
                    this.setState({ name: e.target.value })
                  }}
                  size="mini"
                />
              </Form.Field>
            </Form>
          </Modal.Description>
        </Modal.Content>
        <Modal.Actions>
          <Button className="ui negative button" role="button" onClick={this.props.toggleVisible}>
            Cancel
          </Button>
          <Button primary={true} content="Submit" form="submit-form" />
        </Modal.Actions>
      </Modal>
    )
  }

  private setSelectedCategoryId = (e: any, { value }: { value: string }) => {
    this.setState({ category: value })
  }
}

export default DefineDocumentTypeModal
