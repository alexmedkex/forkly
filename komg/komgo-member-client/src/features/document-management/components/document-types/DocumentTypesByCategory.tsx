import * as React from 'react'
import { List, Accordion, AccordionPanelProps, Checkbox, Icon } from 'semantic-ui-react'
import styled from 'styled-components'
import { indigo, darkViolet } from '../../../../styles/colors'

import { DocumentType, Category } from '../../../document-management/store/types'
import { groupBy } from '../../../document-management/components/documents/my-documents/toMap'

import { PanelCheckboxState } from '../templates/DocumentRequestModal'

export interface Props {
  documentTypes: DocumentType[]
  categories: Category[]
  selectedDocumentTypesByCategory: Map<string, DocumentType[]>
  panelCheckboxState: Map<string, PanelCheckboxState>
  markAsDisabled: (documentTypeName: string) => { isDisabled: boolean; disabledText: string }
  isSelected(documentType: DocumentType): boolean
  onPanelClick(category: Category): void
  onListItemClick(documentType: DocumentType): void
}

class DocumentTypesByCategory extends React.Component<Props> {
  render() {
    return (
      <StyledAccordion
        exclusive={false}
        fluid={true}
        panels={this.props.categories.map(this.categoryToAccordionPanel)}
      />
    )
  }

  private categoryToAccordionPanel = (category: Category): AccordionPanelProps => {
    return {
      key: category.id,
      name: category.name,
      title: { content: this.categoryToPanelTitle(category) },
      content: { content: this.categoryToList(category) }
    }
  }

  private categoryToPanelTitle = (category: Category) => {
    return (
      <>
        <StyledIcon name="chevron down" />
        <StyledCheckbox
          key={category.id}
          floated="left"
          label={category.name}
          checked={this.props.panelCheckboxState.get(category.id) === 1}
          indeterminate={this.props.panelCheckboxState.get(category.id) === 0}
          onClick={(e: React.MouseEvent) => {
            e.preventDefault()
            e.stopPropagation()
            this.props.onPanelClick(category)
          }}
        />
      </>
    )
  }

  private categoryToList = (category: Category) => {
    const groupByCategoryId = (docType: DocumentType) => docType.category.id
    const documentTypesForThisCategory = groupBy(this.props.documentTypes, groupByCategoryId).get(category.id)
    if (!(documentTypesForThisCategory && documentTypesForThisCategory.length)) {
      return null
    }

    return (
      <List
        selection={true}
        name={<b>category.name</b>}
        items={documentTypesForThisCategory.map(this.documentTypeToListItem)}
      />
    )
  }

  private documentTypeToListItem = (documentType: DocumentType) => {
    // default values
    let disabledText: string = ''
    let isDisabled: boolean = false

    // override defaults if markAsDisabled function is provided
    if (this.props.markAsDisabled) {
      const result = this.props.markAsDisabled(documentType.name)
      if (result.isDisabled) {
        disabledText = result.disabledText
        isDisabled = result.isDisabled
      }
    }
    return (
      <List.Item style={{ padding: '7px 2.5em' }}>
        <List.Content key={documentType.id} floated="left">
          <Checkbox
            disabled={false} // We are allowing to re-request same document several times
            label={documentType.name}
            onClick={() => this.props.onListItemClick(documentType)}
            checked={this.props.isSelected(documentType)}
          />
        </List.Content>
        <List.Content key={documentType.id} floated="right">
          <p className="disabled" hidden={!isDisabled}>
            {disabledText}
          </p>
        </List.Content>
      </List.Item>
    )
  }
}

export const StyledAccordion = styled(Accordion)`
  &&& {
    .komgo-accordion-title {
      background-color: ${indigo} !important;
      border-radius: 5px;
      border: 1px solid ${darkViolet};
      padding: 12px 30px;
      height: 45px;
      color: white;
    }

    .title .dropdown.icon {
      display: none;
    }

    .active .komgo-accordion-title {
      border-radius: 5px 5px 0px 0px;
      margin-bottom: -14px;
    }

    i {
      color: white;
      margin-top: 2px;
    }
  }
`

const StyledCheckbox = styled(Checkbox)`
  &&& {
    padding-left: 1em;
    label {
      font-weight: bold;
    }
    .dropdown.icon {
      display: none;
    }
  }
`

const StyledIcon = styled(Icon)`
  &&& {
    width: 16px;
  }
`

export default DocumentTypesByCategory
