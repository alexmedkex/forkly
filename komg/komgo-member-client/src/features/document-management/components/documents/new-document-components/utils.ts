import { HasId, HasName } from '../../../store/types'

export const hasIdAndNameToOption = (collection: Array<HasId & HasName>): Array<{ key: string }> => {
  const originalCollection = collection.map(el => {
    return { key: el.id, text: el.name, value: el.id }
  })
  return originalCollection.reduce((unique: Array<{ key: string }>, o: { key: string }) => {
    if (!unique.some((obj: { key: string }) => obj.key === o.key)) {
      unique.push(o)
    }
    return unique
  }, [])
}
