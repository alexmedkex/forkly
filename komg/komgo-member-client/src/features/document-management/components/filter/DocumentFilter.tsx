import * as React from 'react'
import { Category, DocumentType, DocumentListFilter } from '../../store/types'
import {
  FilterWrapper,
  buildFilterItem,
  CheckboxFilter,
  CheckboxTreeFilter,
  IFilterItemBase,
  ICheckboxOptionGroup,
  ICheckboxOption
} from '@komgo/ui-components'
import { Counterparty } from '../../../counterparties/store/types'
import { getDocumentTypeFilterOptions } from './utils'

export interface IDocumentFilterProps {
  categories: Category[]
  counterparties: Counterparty[]
  types: DocumentType[]
  filter: DocumentListFilter
  disabled?: boolean
  onChange(filter: DocumentListFilter): void
}

export class DocumentFilter extends React.Component<IDocumentFilterProps> {
  handleApply = (filter: DocumentListFilter) => {
    this.props.onChange(filter)
  }

  handleClear = () => {
    this.props.onChange(null)
  }

  buildFilters(
    types: ICheckboxOptionGroup[],
    counterparties: ICheckboxOption[]
  ): Array<IFilterItemBase<DocumentListFilter>> {
    return [
      buildFilterItem(
        'Type',
        'type',
        (filter: DocumentListFilter) => filter.type || [],
        props => <CheckboxTreeFilter options={types} {...props} />,
        (filters: string[]) => (filters ? filters.length : 0)
      ),
      buildFilterItem(
        'Shared with',
        'sharedWith',
        (filter: DocumentListFilter) => filter.sharedWith || [],
        props => <CheckboxFilter {...props} items={counterparties} />,
        (filters: string[]) => (filters ? filters.length : 0)
      )
    ]
  }

  getTypes() {
    return this.props.categories && this.props.types
      ? getDocumentTypeFilterOptions(this.props.categories, this.props.types)
      : []
  }

  getCounterparties(): ICheckboxOption[] {
    return this.props.counterparties
      ? this.props.counterparties.map(c => ({
          value: c.staticId,
          name: c.x500Name.CN
        }))
      : []
  }

  render() {
    const filters = this.buildFilters(this.getTypes(), this.getCounterparties())

    return (
      <FilterWrapper
        filters={filters}
        clearedFilters={{ type: [], sharedWith: [] }}
        appliedFilters={this.props.filter || {}}
        onApply={this.handleApply}
        onClear={this.handleClear}
        disabled={this.props.disabled}
      />
    )
  }
}
