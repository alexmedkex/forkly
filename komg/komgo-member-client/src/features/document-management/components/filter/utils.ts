import { Category, DocumentType } from '../../index'
import { sortCategories, sortDocumentTypes } from '../../utils/sortingHelper'
import { categoryToColor } from '../documents/document-library/categoryToColor'

export const getDocumentTypeFilterOptions = (categories: Category[], types: DocumentType[]) => {
  return sortCategories(categories).map(c => ({
    name: c.name,
    value: c.id,
    display: {
      color: categoryToColor[c.id]
    },
    items: sortDocumentTypes(types)
      .filter(t => t.category.id === c.id)
      .map(t => ({
        name: t.name,
        value: t.id
      }))
  }))
}
