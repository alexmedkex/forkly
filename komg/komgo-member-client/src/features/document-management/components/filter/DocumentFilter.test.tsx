import * as React from 'react'
import { shallow } from 'enzyme'
import * as renderer from 'react-test-renderer'

import { DocumentFilter, IDocumentFilterProps } from './DocumentFilter'
import { mockCategories } from '../../store/categories/mock-data'
import { mockDocumentTypes } from '../../store/document-types/mock-data'
import { fakeCounterparty } from '../../../letter-of-credit-legacy/utils/faker'
import { DocumentListFilter } from '../../store'

describe('DocumentFilter', () => {
  const defaultProps: IDocumentFilterProps = {
    categories: mockCategories,
    types: mockDocumentTypes,
    counterparties: [fakeCounterparty()],
    filter: {},
    onChange: jest.fn()
  }

  it('should match default snapshot', () => {
    expect(renderer.create(<DocumentFilter {...defaultProps} />).toJSON()).toMatchSnapshot()
  })

  it('should send appropriate filter props to filter wrapper - default filter value', () => {
    const wrapper = shallow(<DocumentFilter {...defaultProps} />)

    const filters = wrapper.prop('filters')

    expect(filters[0].label).toBe('Type')
    expect(filters[0].filterKey).toBe('type')
    expect(filters[0].data({})).toEqual([])
    expect(filters[0].content({})).toMatchSnapshot()
    expect(filters[0].calculateApplied([])).toBe(0)

    expect(filters[1].label).toBe('Shared with')
    expect(filters[1].filterKey).toBe('sharedWith')
    expect(filters[1].data({})).toEqual([])
    expect(filters[1].content({})).toMatchSnapshot()
    expect(filters[1].calculateApplied([])).toBe(0)
  })

  it('should send appropriate filter props when filter is active', () => {
    const filter = { type: ['certificate-of-incorporation'], sharedWith: [] }
    const wrapper = shallow(
      <DocumentFilter {...defaultProps} filter={{ type: ['certificate-of-incorporation'], sharedWith: [] }} />
    )

    const filters = wrapper.prop('filters')

    expect(filters[0].data(filter)).toEqual(['certificate-of-incorporation'])
    expect(filters[1].data(filter)).toEqual([])
  })

  it('should call onChange with appropriate props when handleClear is called', () => {
    const wrapper = shallow(<DocumentFilter {...defaultProps} />)

    const instance = wrapper.instance() as DocumentFilter

    instance.handleClear()

    expect(defaultProps.onChange).toHaveBeenLastCalledWith(null)
  })

  it('should call onChange with appropriate props when handleApply is called', () => {
    const wrapper = shallow(<DocumentFilter {...defaultProps} />)

    const instance = wrapper.instance() as DocumentFilter

    const filter: DocumentListFilter = {
      type: ['1234']
    }

    instance.handleApply(filter)

    expect(defaultProps.onChange).toHaveBeenLastCalledWith(filter)
  })
})
