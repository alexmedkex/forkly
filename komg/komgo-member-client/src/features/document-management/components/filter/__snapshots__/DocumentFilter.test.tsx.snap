// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentFilter should match default snapshot 1`] = `
.c1 {
  margin-right: 15px;
  vertical-align: text-bottom;
}

.c0 {
  position: relative;
  display: inline-block;
}

<div
  className="c0"
>
  <button
    className="ui button"
    data-test-id="filter-button"
    onClick={[Function]}
    role="button"
  >
    <img
      className="c1"
      src="data:image/svg+xml;base64,PD94bWwgdmVyc2lvbj0iMS4wIiBlbmNvZGluZz0iVVRGLTgiPz4KPHN2ZyB3aWR0aD0iMTZweCIgaGVpZ2h0PSIxNnB4IiB2aWV3Qm94PSIwIDAgMTYgMTYiIHZlcnNpb249IjEuMSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB4bWxuczp4bGluaz0iaHR0cDovL3d3dy53My5vcmcvMTk5OS94bGluayI+CiAgICA8IS0tIEdlbmVyYXRvcjogU2tldGNoIDU2LjIgKDgxNjcyKSAtIGh0dHBzOi8vc2tldGNoLmNvbSAtLT4KICAgIDx0aXRsZT5TaGFwZTwvdGl0bGU+CiAgICA8ZGVzYz5DcmVhdGVkIHdpdGggU2tldGNoLjwvZGVzYz4KICAgIDxnIGlkPSJQYWdlLTEiIHN0cm9rZT0ibm9uZSIgc3Ryb2tlLXdpZHRoPSIxIiBmaWxsPSJub25lIiBmaWxsLXJ1bGU9ImV2ZW5vZGQiPgogICAgICAgIDxwYXRoIGQ9Ik0wLDEyLjQ0NDQ0NDQgTDAsMTQuMjIyMjIyMiBMNS4zMzMzMzMzMywxNC4yMjIyMjIyIEw1LjMzMzMzMzMzLDEyLjQ0NDQ0NDQgTDAsMTIuNDQ0NDQ0NCBMMCwxMi40NDQ0NDQ0IFogTTAsMS43Nzc3Nzc3OCBMMCwzLjU1NTU1NTU2IEw4Ljg4ODg4ODg5LDMuNTU1NTU1NTYgTDguODg4ODg4ODksMS43Nzc3Nzc3OCBMMCwxLjc3Nzc3Nzc4IEwwLDEuNzc3Nzc3NzggWiBNOC44ODg4ODg4OSwxNiBMOC44ODg4ODg4OSwxNC4yMjIyMjIyIEwxNiwxNC4yMjIyMjIyIEwxNiwxMi40NDQ0NDQ0IEw4Ljg4ODg4ODg5LDEyLjQ0NDQ0NDQgTDguODg4ODg4ODksMTAuNjY2NjY2NyBMNy4xMTExMTExMSwxMC42NjY2NjY3IEw3LjExMTExMTExLDE2IEw4Ljg4ODg4ODg5LDE2IEw4Ljg4ODg4ODg5LDE2IFogTTMuNTU1NTU1NTYsNS4zMzMzMzMzMyBMMy41NTU1NTU1Niw3LjExMTExMTExIEwwLDcuMTExMTExMTEgTDAsOC44ODg4ODg4OSBMMy41NTU1NTU1Niw4Ljg4ODg4ODg5IEwzLjU1NTU1NTU2LDEwLjY2NjY2NjcgTDUuMzMzMzMzMzMsMTAuNjY2NjY2NyBMNS4zMzMzMzMzMyw1LjMzMzMzMzMzIEwzLjU1NTU1NTU2LDUuMzMzMzMzMzMgTDMuNTU1NTU1NTYsNS4zMzMzMzMzMyBaIE0xNiw4Ljg4ODg4ODg5IEwxNiw3LjExMTExMTExIEw3LjExMTExMTExLDcuMTExMTExMTEgTDcuMTExMTExMTEsOC44ODg4ODg4OSBMMTYsOC44ODg4ODg4OSBMMTYsOC44ODg4ODg4OSBaIE0xMC42NjY2NjY3LDUuMzMzMzMzMzMgTDEyLjQ0NDQ0NDQsNS4zMzMzMzMzMyBMMTIuNDQ0NDQ0NCwzLjU1NTU1NTU2IEwxNiwzLjU1NTU1NTU2IEwxNiwxLjc3Nzc3Nzc4IEwxMi40NDQ0NDQ0LDEuNzc3Nzc3NzggTDEyLjQ0NDQ0NDQsMCBMMTAuNjY2NjY2NywwIEwxMC42NjY2NjY3LDUuMzMzMzMzMzMgWiIgaWQ9IlNoYXBlIiBmaWxsPSIjNUQ3NjhGIiBmaWxsLXJ1bGU9Im5vbnplcm8iPjwvcGF0aD4KICAgIDwvZz4KPC9zdmc+"
    />
    Filters
  </button>
</div>
`;

exports[`DocumentFilter should send appropriate filter props to filter wrapper - default filter value 1`] = `
<CheckboxTreeFilter
  options={
    Array [
      Object {
        "display": Object {
          "color": "rgba(187,134,252)",
        },
        "items": Array [],
        "name": "Banking documents",
        "value": "banking-documents",
      },
      Object {
        "display": Object {
          "color": "rgba(132,63,255)",
        },
        "items": Array [],
        "name": "Business description",
        "value": "business-description",
      },
      Object {
        "display": Object {
          "color": "rgba(63,81,181)",
        },
        "items": Array [
          Object {
            "name": "Certificate of Incorporation",
            "value": "certificate-of-incorporation",
          },
        ],
        "name": "Company Details",
        "value": "company-details",
      },
      Object {
        "display": Object {
          "color": "rgba(0,179,188)",
        },
        "items": Array [
          Object {
            "name": "List of Directors",
            "value": "list-of-directors",
          },
        ],
        "name": "Management and directors",
        "value": "management-and-directors",
      },
      Object {
        "display": Object {
          "color": undefined,
        },
        "items": Array [],
        "name": "Miscellaneous",
        "value": "miscellaneous",
      },
      Object {
        "display": Object {
          "color": "rgba(0,134,124)",
        },
        "items": Array [],
        "name": "Regulation/Compliance",
        "value": "regulation-and-compliance",
      },
      Object {
        "display": Object {
          "color": "rgba(68,33,133)",
        },
        "items": Array [
          Object {
            "name": "Passports of UBOs",
            "value": "passports-of-ubos",
          },
        ],
        "name": "Shareholders - Ultimate Beneficiary Owners (UBOs)",
        "value": "shareholders",
      },
    ]
  }
/>
`;

exports[`DocumentFilter should send appropriate filter props to filter wrapper - default filter value 2`] = `
<CheckboxFilter
  items={
    Array [
      Object {
        "name": "Applicant Name",
        "value": "cf63c1f8-1165-4c94-a8f8-9252eb4f0016",
      },
    ]
  }
/>
`;
